modules/mill-codegen-plugin/src/smithy4s/codegen/mill/Smithy4sExternalModule.scala

```diff
@@ -21,81 +21,48 @@ import mill.define.ExternalModule
 import mill.define.Target
 import coursier.maven.MavenRepository
 import smithy4s.codegen.SmithyBuildJson
-import smithy4s.codegen.SmithyBuildData
 import mill.define.Command
 import mill.api.PathRef
 
 object Smithy4sExternalModule extends ExternalModule {
   lazy val millDiscover = mill.define.Discover[this.type]
 
-  def generateSmithyBuildJson(ev: Evaluator): Command[PathRef] =
-    Target.command {
-      val rootPath = ev.rootModule.millModuleBasePath.value
-      val s4sModules = ev.rootModule.millInternal.modules
-        .collect { case s: Smithy4sModule => s }
+  def generateSmithyBuildJson(ev: Evaluator): Command[PathRef] = {
+    val rootPath = ev.rootModule.millModuleBasePath.value
+    val s4sModules = ev.rootModule.millInternal.modules
+      .collect { case s: Smithy4sModule => s }
 
-      def extractImports(m: Smithy4sModule): Seq[String] = {
-        ev
-          .evalOrThrow()(m.smithy4sInputDirs)
-          .map(p => p.path.relativeTo(rootPath))
-          .map(_.toString)
-          .distinct
-      }
+    val depsTask = Target
```

Comment:
I had a similar approach at first and I rewrote to avoid traversing the s4smodule seq multiple time (like I did in sbt)

Reply:
Yeah, the approach you had was similar to chaining IO using `unsafeRunSync`, which ain't great. 

I can't comment much on whether the SBT approach is idiomatic because I don't know SBT as well 😅